app/models/metric.rb

```diff
@@ -1,3 +1,37 @@
+# Note on the database layout of `metrics` and `metrics_rollups`
+#
+# In the database, the `metrics` table leverages [PostgreSQL table inheritance]
+# and [PostgreSQL table partioning].
+#
+# Table inheritance is used to create child tables with the same structure as
+# the parent.  As columns are added or removed from the parent table, they are
+# automatically added and removed from the child tables.  The `metrics` table,
+# which is used for "realtime" data, has 24 child tables, one per hour.  The
+# `metrics_rollups` table, which is used for long term "rollup" data, has 12
+# child tables, one per month.
+#
+# Table partitioning is used to divert inserts and deletes from the parent table
+# to a specific child table, and to allow queries to search across the various
+# child tables and return the results as if they were in a single table.  This
+# is accomplished using triggers on the parent table, which look at the hour of
+# the timestamp for `metrics` or the month of the timestamp for
+# `metric_rollups`, and then diverting to the corresponding child table.
+# Additionally, a shared sequence is used from the parent table for the primary
+# key, so that when queries blend the rows together the primary key constraint
+# is satisfied.
```

Comment:
a) We may want to try and avoid the word "this"
b) I think this paragraph is great - it introduces the key concepts that enable our sharding.
c) Maybe reword a little? I tried to keep it close to what you wrote.

When data is inserted into the parent table, triggers divert the data to the child tables.
When data is selected or deleted from the parent table, table inheritance performs the functions on all the child tables. The queries will only be performed on a subset of the child tables if the constraints are properly met.
Additionally, the sequence from the parent table is shared by the child table so the primary key constraint is satisfied even when records are directly inserted into the child tables.

Reply:
I think what you are suggesting here feels like it should be in a list form, and needs some kind of lead in paragraph:

> Table partitioning is used to redirect database operations to the sub tables.  For each operation, the operations taken by the code/DB are as follows
> * When data is inserted into the parent table, triggers divert the data to the child tables.
> * When data is selected or deleted from the parent table, table inheritance performs the functions on all the child tables. The queries will only be performed on a subset of the child tables if the constraints are properly met.
> 
> Additionally, the sequence from the parent table is shared by the child table so the primary key constraint is satisfied even when records are directly inserted into the child tables.

But even then, that is a lot of extra words to say the same thing as the above, so I really don't agree with the suggested change here...